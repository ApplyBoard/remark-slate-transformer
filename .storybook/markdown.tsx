import React, { useCallback, useMemo, useState } from "react";
import unified from "unified";
import markdown from "remark-parse";
import remark2rehype from "remark-rehype";
import html from "rehype-stringify";

const processor = unified()
  .use(markdown, { commonmark: true })
  .use(remark2rehype)
  .use(html);

const style: React.CSSProperties = {
  flex: 1,
  margin: 10,
};

type Props = {
  initialValue: string;
};

export default ({ initialValue }: Props) => {
  const html = useMemo(
    () => ({ __html: processor.processSync(initialValue).contents as string }),
    [initialValue]
  );
  return (
    <div style={style}>
      <div dangerouslySetInnerHTML={html} />
    </div>
  );
};
