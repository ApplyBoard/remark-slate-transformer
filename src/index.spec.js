import fs from "fs";
import path from "path";
import unified from "unified";
import markdown from "remark-parse";
import stringify from "remark-stringify";
import { remarkToSlate, slateToRemark, remarkToSlateLegacy } from "./";
import { Value } from "slate0.47";

const FIXTURE_PATH = "../fixture";

describe("e2e", () => {
  const toSlateProcessor = unified()
    .use(markdown, { commonmark: true })
    .use(remarkToSlate);
  const toRemarkProcessor = unified().use(slateToRemark).use(stringify);

  const fixtureDir = path.join(__dirname, FIXTURE_PATH);
  const filenames = fs.readdirSync(fixtureDir);
  filenames.forEach((filename) => {
    it(filename, () => {
      const slateNodes = toSlateProcessor.processSync(
        fs.readFileSync(path.join(fixtureDir, filename))
      ).result;
      expect(slateNodes).toMatchSnapshot();

      const mdastTree = toRemarkProcessor.runSync({
        type: "root",
        children: slateNodes,
      });
      expect(mdastTree).toMatchSnapshot();

      const text = toRemarkProcessor.stringify(mdastTree);
      expect(text).toMatchSnapshot();
    });
  });
});

describe("e2e legacy", () => {
  const toSlateProcessor = unified()
    .use(markdown, { commonmark: true })
    .use(remarkToSlateLegacy);
  // const toRemarkProcessor = unified().use(slateToRemark).use(stringify);

  const fixtureDir = path.join(__dirname, FIXTURE_PATH);
  const filenames = fs.readdirSync(fixtureDir);
  filenames.forEach((filename) => {
    it(filename, () => {
      const slateNodes = toSlateProcessor.processSync(
        fs.readFileSync(path.join(fixtureDir, filename))
      ).result;
      expect(slateNodes).toMatchSnapshot();
      expect(Value.fromJSON(slateNodes)).toMatchSnapshot();

      // const mdastTree = toRemarkProcessor.runSync({
      //   type: "root",
      //   children: slateNodes,
      // });
      // expect(mdastTree).toMatchSnapshot();

      // const text = toRemarkProcessor.stringify(mdastTree);
      // expect(text).toMatchSnapshot();
    });
  });
});
