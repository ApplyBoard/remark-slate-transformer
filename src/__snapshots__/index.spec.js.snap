// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin article 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "heading 1",
      },
    ],
    "depth": 1,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "heading 2",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "heading 3",
      },
    ],
    "depth": 3,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "heading 4",
      },
    ],
    "depth": 4,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "heading 5",
      },
    ],
    "depth": 5,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "heading 6",
      },
    ],
    "depth": 6,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "Test",
      },
    ],
    "depth": 1,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "aaaaaaaaaaaaaaaaa
",
      },
      Object {
        "children": Array [
          Object {
            "text": "bold",
          },
        ],
        "type": "strong",
      },
      Object {
        "text": "
",
      },
      Object {
        "children": Array [
          Object {
            "text": "emphasis",
          },
        ],
        "type": "emphasis",
      },
      Object {
        "text": "
",
      },
      Object {
        "children": Array [
          Object {
            "text": "delete",
          },
        ],
        "type": "delete",
      },
      Object {
        "text": "
",
      },
      Object {
        "children": Array [
          Object {
            "text": "inline code",
          },
        ],
        "type": "inlineCode",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "list",
              },
            ],
            "type": "paragraph",
          },
          Object {
            "children": Array [
              Object {
                "checked": null,
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "list",
                      },
                    ],
                    "type": "paragraph",
                  },
                  Object {
                    "children": Array [
                      Object {
                        "checked": null,
                        "children": Array [
                          Object {
                            "children": Array [
                              Object {
                                "text": "list",
                              },
                            ],
                            "type": "paragraph",
                          },
                        ],
                        "spread": false,
                        "type": "listItem",
                      },
                    ],
                    "ordered": false,
                    "spread": false,
                    "start": null,
                    "type": "list",
                  },
                ],
                "spread": false,
                "type": "listItem",
              },
              Object {
                "checked": null,
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "list",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "spread": false,
                "type": "listItem",
              },
              Object {
                "checked": null,
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "list",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "spread": false,
                "type": "listItem",
              },
            ],
            "ordered": false,
            "spread": false,
            "start": null,
            "type": "list",
          },
        ],
        "spread": true,
        "type": "listItem",
      },
      Object {
        "checked": false,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "checked",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
      Object {
        "checked": true,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "not checked",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
    ],
    "ordered": false,
    "spread": true,
    "start": null,
    "type": "list",
  },
  Object {
    "children": Array [
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "ordered list",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "ordered list",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "ordered list",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
    ],
    "ordered": true,
    "spread": false,
    "start": 1,
    "type": "list",
  },
  Object {
    "children": Array [
      Object {
        "text": "bbbb",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "link to GitHub.com",
          },
        ],
        "title": null,
        "type": "link",
        "url": "https://github.com/",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "alt": "image",
        "children": Array [
          Object {
            "text": "",
          },
        ],
        "title": "image",
        "type": "image",
        "url": "https://github.githubassets.com/images/modules/logos_page/Octocat.png",
        "void": true,
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "quote
quote
quote
quote",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "blockquote",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "quote",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "quoted quote",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "blockquote",
      },
    ],
    "type": "blockquote",
  },
  Object {
    "align": Array [
      "left",
      "right",
      "center",
    ],
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Left align",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "Right align",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "Center align",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "This",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "This",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "This",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "column",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "column",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "column",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "will",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "will",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "will",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "be",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "be",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "be",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "left",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "right",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "center",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "aligned",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "aligned",
              },
            ],
            "type": "tableCell",
          },
          Object {
            "children": Array [
              Object {
                "text": "aligned",
              },
            ],
            "type": "tableCell",
          },
        ],
        "type": "tableRow",
      },
    ],
    "type": "table",
  },
  Object {
    "children": Array [
      Object {
        "text": "cccc",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "type": "thematicBreak",
    "void": true,
  },
  Object {
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "type": "thematicBreak",
    "void": true,
  },
  Object {
    "children": Array [
      Object {
        "text": "@font-face {
  font-family: Chunkfive;
  src: url(\\"Chunkfive.otf\\");
}

body,
.usertext {
  color: #f0f0f0;
  background: #600;
  font-family: Chunkfive, sans;
  --heading-1: 30px/32px Helvetica, sans-serif;
}

@import url(print.css);
@media print {
  a[href^=\\"http\\"]::after {
    content: attr(href);
  }
}",
      },
    ],
    "lang": "css",
    "meta": null,
    "type": "code",
  },
  Object {
    "children": Array [
      Object {
        "text": "function $initHighlight(block, cls) {
  try {
    if (cls.search(/\\\\bno\\\\-highlight\\\\b/) != -1)
      return process(block, true, 0x0F) +
             \` class=\\"\${cls}\\"\`;
  } catch (e) {
    /* handle exception */
  }
  for (var i = 0 / 2; i < classes.length; i++) {
    if (checkCondition(classes[i]) === undefined)
      console.log('undefined');
  }

  return (
    <div>
      <web-component>{block}</web-component>
    </div>
  )
}

export  $initHighlight;",
      },
    ],
    "lang": "javascript",
    "meta": null,
    "type": "code",
  },
  Object {
    "children": Array [
      Object {
        "text": "class MyClass {
  public static myValue: string;
  constructor(init: string) {
    this.myValue = init;
  }
}
import fs = require(\\"fs\\");
module MyModule {
  export interface MyInterface extends Other {
    myProperty: any;
  }
}
declare magicNumber number;
myArray.forEach(() => { }); // fat arrow syntax",
      },
    ],
    "lang": "typescript",
    "meta": null,
    "type": "code",
  },
]
`;

exports[`plugin headings 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "Alpha",
      },
    ],
    "depth": 1,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "aaaa",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "Bravo",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "bbbb",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "Charlie",
      },
    ],
    "depth": 3,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "cccc",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "Delta",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "text": "dddd",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`plugin plain text 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      },
    ],
    "type": "paragraph",
  },
  Object {
    "children": Array [
      Object {
        "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      },
    ],
    "type": "paragraph",
  },
]
`;
